src/qwed_new/providers/openai_direct.py

```
@@ -0,0 +1,252 @@
+from qwed_new.providers.base import LLMProvider
+
+logger = logging.getLogger(__name__)
+
+
+class OpenAIDirectProvider(LLMProvider):
+    """
+    Provider for direct OpenAI API (gpt-4o, gpt-4o-mini, etc).
+    Uses function calling (tools) for structured output.
+    """
+
+    def __init__(self, api_key: str = None, model: str = None):
```

Ruff (0.15.6)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)